pages/_document.tsx

```diff
@@ -64,6 +64,16 @@ class MyDocument extends Document {
   ): Promise<DocumentInitialProps & MyDocumentProps> {
     invariant(context.req)
     const jwt = context.req?.cookies?.[COOKIE_JWT_TOKEN] || null
+
+    // set cache control header
+    context.res?.setHeader(
+      'Cache-Control',
+      `${
+        // make the cache private if user is logged in
+        jwt ? 'private' : 'public'
+      }, max-age=10, stale-while-revalidate=10`,
```

Comment:
We could change the value to something like this:
```suggestion
      }, max-age=1, stale-while-revalidate=60`,
```
This will basically refresh the cache every 1sec but allow to render the staled cached up to 60sec to give enough time to Vercel to refresh the cache in the background. So if there are thousands of users, the cache will be refresh every 1 second + the time to generate the page.